csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.cs

```diff
@@ -993,9 +998,171 @@ public bool IsTensor
     }
 
 
-    internal class ModelMetadata
+    public class ModelMetadata
```

Comment:
>   public class ModelMetadata [](start = 1, length = 29)

You have chosen to implement the class in a way that queries everything. Another way to imlpement it is to query things on demand, only when needed. This is the way OrtMemoryInfo is implemented so the user pays only for what is needed.
Have you considered doing it this way and not to cache anything in managed memory? What is the use pattern here?

Reply:
I thought about the "pay for what you want" approach but as synced offline - I feel that it doesn't provide too much value as folks are most likely going to require most of the metadata  anyway and we would have to cache the model metadata native handle for the lifetime of the instance and dispose it at the end - I think this approach works for this kind of a class.